obs/project.go

```diff
@@ -91,3 +96,110 @@ type RepositoryPath struct {
 	Project    string `json:"project" xml:"project,attr"`
 	Repository string `json:"repository" xml:"repository,attr"`
 }
+
+type Status struct {
+	XMLName xml.Name `json:"status" xml:"status"`
+	Code    string   `json:"code" xml:"code,attr"`
+	Summary string   `json:"summary" xml:"summary"`
+}
+
+type APIError struct {
+	HTTPStatusCode int
+	XMLStatusCode  string
+	Message        string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("HTTP status %d: %s (%s)", e.HTTPStatusCode, e.XMLStatusCode, e.Message)
+}
+
+func (c Client) CreateProject(project *Project) error {
+	xmlData, err := xml.MarshalIndent(project, "", " ")
+	if err != nil {
+		return err
+	}
+
+	urlPath := path.Join(c.APIURL, "source", project.Name, "_meta")
+	req, err := http.NewRequest("PUT", urlPath, bytes.NewBuffer(xmlData))
+	if err != nil {
+		return &APIError{
+			HTTPStatusCode: http.StatusInternalServerError,
```

Comment:
We shouldn't use InternalServerError here. This function returns an error if an user made the said error (e.g. invalid URL). That's for sure not a server error and the error code should reflect that. In this case, maybe we can just set 0 here and provide details in `Message`.

Reply:
Resolved